HW2/P2/P2.txt

```diff
@@ -0,0 +1,21 @@
+1. Time comparison
+
+(1) Uncorrelated (random)
+Serial uncorrelated: 0.255743026733 seconds
+Fine grained uncorrelated: 6.3752450943 seconds
+Medium grained uncorrelated: 6.8746650219 seconds
+
+(2) Correlated (N=10)
+Serial correlated: 0.330317974091 seconds
+Fine grained correlated: 6.20790791512 seconds
+Medium grained correlated: 5.56719088554 seconds
+
+When data exchange was correlated, we could learn that Medium grained locks were showing better performance compared to the Fine grained locks. This is because, since there are more threads sharing the locks, less overhead for waiting those locks would result such better performance.
```

Comment:
Actually, since there are less locks, there should be MORE waiting (you're more likely to run into a lock conflict).